packages/atomic/src/components/commerce/atomic-commerce-recommendation-list/e2e/atomic-commerce-recommendation-list.e2e.ts

```diff
@@ -0,0 +1,103 @@
+  });
+
+  test('should have placeholders', async ({recommendationList}) => {
+    await expect(recommendationList.placeholder.first()).toBeVisible();
+  });
+});
+
+test.describe('after query is loaded', () => {
+  test.beforeEach(async ({recommendationList}) => {
+    await recommendationList.load({story: 'default'});
+    await recommendationList.hydrated.waitFor();
+  });
+
+  test('should be a11y compliant', async ({makeAxeBuilder}) => {
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations.length).toEqual(0);
+  });
+
+  test('should have recommendations', async ({recommendationList}) => {
+    await expect(recommendationList.recommendation.first()).toBeVisible();
+  });
+});
+
+test.describe('with a full result template', () => {
+  test.beforeEach(async ({recommendationList}) => {
+    await recommendationList.load({story: 'with-full-template'});
+    await recommendationList.hydrated.waitFor();
+  });
+
+  test('should be a11y compliant', async ({makeAxeBuilder}) => {
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations.length).toEqual(0);
+  });
+
+  test('should have recommendations', async ({recommendationList}) => {
+    await expect(recommendationList.recommendation.first()).toBeVisible();
+  });
+});
+
+test.describe('with a carousel', () => {
+  test.beforeEach(async ({recommendationList}) => {
+    await recommendationList.load({story: 'as-carousel'});
+    await recommendationList.hydrated.waitFor();
+  });
+
+  test('should be a11y compliant', async ({makeAxeBuilder}) => {
+    const accessibilityResults = await makeAxeBuilder().analyze();
+    expect(accessibilityResults.violations.length).toEqual(0);
+  });
+
+  test('should have recommendations', async ({recommendationList}) => {
+    await expect(recommendationList.recommendation.first()).toBeVisible();
+  });
+
+  test('should support going forward and backward', async ({
+    recommendationList,
+  }) => {
+    await recommendationList.nextButton.click();
+    await expect(recommendationList.indicators.nth(1)).toHaveAttribute(
+      'part',
+      'indicator active-indicator'
+    );
+
+    await recommendationList.prevButton.click();
+    await recommendationList.prevButton.click();
+    await expect(recommendationList.indicators.nth(2)).toHaveAttribute(
+      'part',
+      'indicator active-indicator'
+    );
+
+    await recommendationList.nextButton.click();
+    await expect(recommendationList.indicators.nth(0)).toHaveAttribute(
+      'part',
+      'indicator active-indicator'
+    );
+  });
+});
+
+test('with no recommendations returned by the API, should render placeholders', async ({
+  recommendationList,
+  page,
+}) => {
+  await recommendationList.noRecommendations();
+  await recommendationList.load({story: 'default'});
+  await page.waitForTimeout(2000); // Wait for atomic components to render
```

Comment:
As discussed in person, I'm not a fan of adding an arbitrary timeout here. If we take the habit of doing that, our e2e tests will progressively get slower and slower.

Two arguably better alternatives would be to:

1) wait for the aria-live to be hydrated
or
2) render a <Hidden> element instead of rendering nothing when there are no recs

the second option is probably better!

Reply:
went with the second option